app/models/middleware_server.rb

```diff
@@ -33,6 +37,22 @@ def evaluate_alert(alert_id, event)
   end
 
   def in_domain?
-    !middleware_server_group.nil?
+    middleware_server_group
+  end
+
+  # Returns whether a server is immutable or not.
+  #
+  # By default, we define all servers as being immutable, so that power
+  # operations are not allowed, and it is the provider responsability to
+  # reimplement this method if necessary.
+  def immutable?
+    true
+  end
+
+  # Returns whether a server is immutable or not.
+  #
+  # Just a convenience method over #immutable.
+  def mutable?
+    !immutable?
```

Comment:
Not a big fan of this since we already have `if`/`unless` to avoid negating a conditional but I won't hold this up because of it